### PR TITLE
SG-19789 Updates to software Licenses

### PR DESCRIPTION
**Pull request by eshokrgozar**
- Added ordereddict to the requirements.  **edit**: This was needed because I was not running the script in python 2.6. Removed it again
- Added a pip freeze step to the third_party script. 
- Updated the software_credits. 
- Re-installed the third_party packages. 
- Updated six version to match in both the main source and the test dir.
- Updated the third_party upgrade script to make sure it only checks locally installed modules

### CODE REVIEW

**Review by pscadding — CHANGES_REQUESTED** [review]
I think setuptools is missing from your software credits.
Other than that, it looks great!

**Review by pscadding — APPROVED** [review]

File: python/tank_vendor/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+distro==1.4.0
```
Comment (jfboismenu) [inline python/tank_vendor/requirements.txt:1]: Please add copyright notice at the top.

File: tests/python/requirements.txt [hunk 2]
```diff
@@ -14,6 +14,8 @@
 # Use to temporarily replace part of the API or mock return vaues.
 mock==2.0.0
 
+ordereddict==1.1
```
Comment (jfboismenu) [inline tests/python/requirements.txt:17]: Are we actually using it?

**Review by jfboismenu — APPROVED** [review]
Aside from missing a couple of copyright notices, this looks good to me.

**Pull request closed without merging**